### PR TITLE
ObjectType field values needs to be described

### PR DESCRIPTION
**Pull request by ihsandemir**
Added more description of what the ObjectType fields can be.
Some typing fixes are done.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
 
 TypedData does late deserialization of the data only when the get method is called.
 
+This TypedData allows you to retrieve the data type of the underlying binary to be used when being deserialized. This class represents the type of a Hazelcast serializable object. The fields can take the following values:
```
Comment (Serdaro) [inline README.md]: There should be line break after this paragraph, otherwise the numbered list will not be formatted.

File: README.md [hunk 2]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
...
+1. <b>Primitive types</b>: factoryId=-1, classId=-1, typeId is the type id for that primitive as listed in
```
Comment (Serdaro) [inline README.md]: ....is the type ID for that....

File: README.md [hunk 3]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
...
+@link SerializationConstants
```
Comment (Serdaro) [inline README.md]: Is "@link" working as expected?

File: README.md [hunk 4]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
...
+2. <b>Array of primitives</b>: factoryId=-1, classId=-1, typeId is the type id for that array as listed in
```
Comment (Serdaro) [inline README.md]: ....is the type ID for that...

File: README.md [hunk 5]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
...
+@link SerializationConstants
+3. <b>IdentifiedDataSerializable</b>: factory, class and type ids are non-negative values as registered by
```
Comment (Serdaro) [inline README.md]: ....and type IDs are non-negative....

File: README.md [hunk 6]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
...
+the DataSerializableFactory.
+4. <b>Portable</b>: factory, class and type ids are non-negative values as registered by the PortableFactory.
```
Comment (Serdaro) [inline README.md]: ....and type IDs are non-negative...

File: README.md [hunk 7]
```diff
@@ -487,6 +487,17 @@ The TypedData class is a wrapper class for the serialized binary data. It presen
...
+5. <b>Custom serialized objects</b>: factoryId=-1, classId=-1, typeId is the non-negative type id as
```
Comment (Serdaro) [inline README.md]: ....the non-negative type ID as....

Comment (Serdaro) [reply README.md]: and please make "id"s in the code comments below "ID".

Comment (ihsandemir) [reply README.md]: OK, corrected

File: README.md (hunk 6 above)
Comment (ihsandemir) [reply README.md]: OK, changed.

File: README.md (hunk 5 above)
Comment (ihsandemir) [reply README.md]: OK, changed.

File: README.md (hunk 4 above)
Comment (ihsandemir) [reply README.md]: OK, corrected.

File: README.md (hunk 3 above)
Comment (ihsandemir) [reply README.md]: It was not working, I removed it.

File: README.md (hunk 2 above)
Comment (ihsandemir) [reply README.md]: Ok, corrected.

**Comment by ihsandemir** [comment]
verify

**Pull request merged**